lib/resource.js

```diff
@@ -1,170 +1,104 @@
 'use strict';
 
-const request = require('superagent');
-const paginate = require('./paginate');
+import request from 'superagent';
+import { generateUrl } from './urlHelpers';
+import {
+  getMethod,
+  isObject,
+  isFunction,
+} from './utils';
 
 class Resource {
   constructor(api, path, methods, isToken) {
-    let _this         = this;
-    _this._api        = api;
-    _this.methods     = methods;
-    _this.path        = path;
+    this._api    = api;
+    this.methods = methods;
+    this.path    = path;
 
-    _this.init();
+    this.init();
   }
 
   init() {
-    let _this = this;
-
-    _this.methods.forEach(function(item) {
-      let method        = _this.getMethod(item);
-      let resource_type = _this.path || 'id';
-      let params = {
+    this.methods.forEach((item) => {
+      const method = getMethod(item);
+      const params = {
         http_method: 'get', // superagent reads 'get' not 'GET'
         client_method: method
       };
 
-      if (method.match(/retrieve|items|files/i)) {
-        _this[method] = function(parsed_params, callback) {
-          if (_this.isCallbackFunction(callback)) {
-            params.callback = callback;
-          }
-
-          params.options   = parsed_params;
-          params.url       = this.parseUrl(parsed_params);
-          params.scope     = item.scope ? item.scope : null;
-
-          _this.makeRequest(params);
-        };
+      if (method.match(/files|items|watching|watchlist/)) {
+        params.scope = item;
       }
-      else {
-        _this[method] = function(options, callback) {
-          params.options   = options;
-          params.url       = this.parseUrl(options);
-          params.callback  = callback;
-
-          _this.makeRequest(params);
-        };
-      }
-    });
-  }
 
-  parseUrl(params) {
-    if (params && params.collection) {
-      return params.collection;
-    }
+      this[method] = (options, callback) => {
+        params.url       = generateUrl(options, this, params);
+        params.callback  = callback;
 
-    if (params && params.product) {
-      if (!parseInt(params.product, 10)) {
-        return params.product;
-      }
-    }
-
-    if (params && this.path.match(/videos/)) {
-      return params;
-    }
-
-    if (params && this.path.match(/customers/)) {
-      return params;
-    }
-
-    return `${this._api.protocol}${this._api.host}/${this.path}`;
-  }
-
-  getMethod(params) {
-    let _this = this;
-
-    if (_this.isObject(params) && params.method) {
-      return params.method;
-    } else {
-      return params;
-    }
-  }
-
-  needsToken() {
-    return new Date() > this._api.token_expiration;
+        this.createRequestParams(params);
+      };
+    });
   }
 
   getParams(client_method, url, options, scope, type) {
-    let _this  = this;
     let params = {};
```

Comment:
`const params = {};` could be used here instead of `let` if you are seeking to use `const` everywhere possible. Since our eslint config is based on the airbnb one, we probably have `prefer-const` in our linting rules anyway: https://github.com/airbnb/javascript#references

I'll hold off on more comments of this type since we could do a subsequent PR that implements the VHX eslint config: https://www.npmjs.com/package/@vhx/eslint-config-vhx